src/Sentry/Timing.cs

```diff
@@ -0,0 +1,100 @@
+using Sentry.Extensibility;
+using Sentry.Protocol.Metrics;
+
+namespace Sentry;
+
+/// <summary>
+/// Measures the time it takes to run a given code block and emits this as a metric. The <see cref="Timing"/> class is
+/// designed to be used in a <c>using</c> statement.
+/// </summary>
+/// <example>
+/// using (var timing = new Timing("my-operation"))
+/// {
+///     ...
+/// }
+/// </example>
+public class Timing: IDisposable
+{
+    private readonly IHub _hub;
+    private readonly string _key;
+    private readonly MeasurementUnit.Duration _unit;
+    private readonly IDictionary<string, string>? _tags;
+    private readonly Stopwatch _stopwatch = new();
+    private readonly ISpan _span;
+    private readonly DateTime _startTime = DateTime.UtcNow;
+
+    /// <summary>
+    /// Creates a new <see cref="Timing"/> instance.
+    /// </summary>
+    public Timing(string key, MeasurementUnit.Duration unit = MeasurementUnit.Duration.Second,
+        IDictionary<string, string>? tags = null)
+        : this(SentrySdk.CurrentHub, key, unit, tags, stackLevel: 2 /* one for each constructor */)
+    {
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="Timing"/> instance.
+    /// </summary>
+    public Timing(IHub hub, string key, MeasurementUnit.Duration unit = MeasurementUnit.Duration.Second,
+        IDictionary<string, string>? tags = null)
+    : this(hub, key, unit, tags, stackLevel: 2 /* one for each constructor */)
+    {
+    }
+
+    internal Timing(IHub hub, string key, MeasurementUnit.Duration unit, IDictionary<string, string>? tags,
+        int stackLevel)
+    {
+        _hub = hub;
+        _key = key;
+        _unit = unit;
+        _tags = tags;
+        _stopwatch.Start();
+
+        ITransactionTracer? currentTransaction = null;
+        hub.ConfigureScope(s => currentTransaction = s.Transaction);
+        _span = currentTransaction is {} transaction
+            ? transaction.StartChild("metric.timing", key)
+            : hub.StartTransaction("metric.timing", key);
```

Comment:
Just spotted this (sorry if it was here from the start). But are we creating a transaction named `metric.timing` that we don't attach to the scope to hold this metric? Curious if we do that in Python too

Reply:
Yeah that's lifted from here:
https://github.com/getsentry/sentry-python/blob/67c963d9c8d5e7e9de6347aee0edcf0c58d9fb24/sentry_sdk/metrics.py/#L790

Python doesn't do anything special with the scope but is that due to fundamental differences between the Python SDK and the .NET SDK? Should we be setting the scope transaction in .NET?

Reply:
> Should we be setting the scope transaction in .NET?
No lets not, it's fine then, we're just creating a local transaction to be a container for the metric. Let see how this works